modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/FieldValue.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.iglu.schemaddl.parquet
+
+import io.circe._
+import cats.implicits._
+import cats.data.ValidatedNel
+
+import java.time.Instant
+import java.time.format.DateTimeFormatter
+import CastError._
+
+/** Run-time value, conforming [[Field]] (type) */
+sealed trait FieldValue extends Product with Serializable
+
+object FieldValue {
+  case object NullValue extends FieldValue
+  case class JsonValue(value: Json) extends FieldValue
+  case class StringValue(value: String) extends FieldValue
+  case class BooleanValue(value: Boolean) extends FieldValue
+  case class IntValue(value: Int) extends FieldValue
+  case class LongValue(value: Long) extends FieldValue
+  case class DoubleValue(value: Double) extends FieldValue
+  case class DecimalValue(value: BigDecimal) extends FieldValue
+  case class TimestampValue(value: java.sql.Timestamp) extends FieldValue
+  case class DateValue(value: java.sql.Date) extends FieldValue
+  case class StructValue(values: List[(String, FieldValue)]) extends FieldValue
```

Comment:
I chose to add a `NamedValue` case class, because I agree the tuple was a bit ugly.

I chose not to add a case class for `Field.Name` or `Decimal.Scale` because........  Uh, hard to explain.  I just didn't feel it added anything to wrap them.  It made it more inconvenient to use in downstream applications, but without much obvious benefit.

I don't feel too strongly about this though, I would make them case classes if you think it improves the API interface of this library.